com.unity.multiplayer.mlapi/Runtime/Core/NetworkUpdateManager.cs

```diff
@@ -4,216 +4,193 @@
 
 using System;
 using System.Text;
+using System.Collections.Generic;
 using UnityEngine;
 using UnityEngine.PlayerLoop;
 using UnityEngine.LowLevel;
 
 namespace MLAPI
 {
     /// <summary>
-    /// Allows one to create their own network update engine
-    /// in the event they have specific processing needs etc.
+    /// NetworkUpdateManager
+    /// External public facing class for the registration of and processing of network updates
     /// </summary>
-    public interface INetworkUpdateEngine
+    public class NetworkUpdateManager
     {
-        void PreUpdate();
-        void PreUpdateRegister(Action updateAction);
-
-        void FixedUpdate();
-        void FixedUpdateRegister(Action updateAction);
-
-        void Update();
-        void UpdateRegister(Action updateAction);
+        public enum NetworkUpdateStages
+        {
+            DEFAULT,        //Will default to the UPDATE stage if no setting was made
+            PREUPDATE,      //Invoked after EarlyUpdate.UnityWebRequestUpdate
+            FIXEDUPDATE,    //Invoked after FixedUpdate.AudioFixedUpdate (prior to any physics being applied or simulated)
+            UPDATE,         //Invoked after PreUpdate.UpdateVideo   (just before the primary Update is invoked)
+            LATEUPDATE      //Invoked after PostLateUpdate.ProcessWebSendMessages (after all updates)
```

Comment:
I'd suggest slight changes:

```
Default,
PreUpdate,
FixedUpdate,
Update,
LateUpdate
```

Reply:
Good catch!  Definitely want everything to be to standard.